crates/rpc/src/v02/method/estimate_fee.rs

```diff
@@ -151,60 +148,136 @@ mod tests {
+                        //     data_offset: felt,   // Offset in the "outer" calldata (BroadcastedInvokeTransactionV1::calldata) to the next contract's calldata
+                        //     data_len: felt,      // Size of the calldata for this contract function call
+                        // }
+                        //
+                        // To see how the above structure is translated to a proper calldata for a single call instance see
+                        // a "preset" implementation of IAccount
+                        // https://github.com/OpenZeppelin/cairo-contracts/blob/4dd04250c55ae8a5bbcb72663c989bb204e8d998/src/openzeppelin/account/presets/Account.cairo#L128
+                        // https://github.com/OpenZeppelin/cairo-contracts/blob/main/src/openzeppelin/account/library.cairo#L239
+                        //
+                        // especially
+                        //
+                        // func _from_call_array_to_call{syscall_ptr: felt*}(
+                        //     call_array_len: felt, call_array: AccountCallArray*, calldata: felt*, calls: Call*
+                        // )
+                        //
+                        // Called contract address, i.e. AccountCallArray::to
+                        CallParam(felt!(
+                            "05a02acdbf218464be3dd787df7a302f71fab586cad5588410ba88b3ed7b3a21"
+                        )),
+                        // Entry point selector for the called contract, i.e. AccountCallArray::selector
+                        CallParam(felt!(
+                            "03d7905601c217734671143d457f0db37f7f8883112abd34b92c4abfeafde0c3"
+                        )),
+                        // Length of the call data for the called contract, i.e. AccountCallArray::data_len
+                        CallParam(felt!("2")),
+                        // Proper calldata for this contract
+                        CallParam(felt!(
+                            "e150b6c2db6ed644483b01685571de46d2045f267d437632b508c19f3eb877"
+                        )),
+                        CallParam(felt!(
+                            "0494196e88ce16bff11180d59f3c75e4ba3475d9fba76249ab5f044bcd25add6"
+                        )),
+                    ],
+                },
+            ))
         }
 
-        fn valid_broadcasted_transaction() -> BroadcastedTransaction {
-            BroadcastedTransaction::Invoke(BroadcastedInvokeTransaction::V0(
-                valid_mainnet_invoke_v0(),
-            ))
+        pub(crate) fn test_storage_with_account() -> (
+            tempfile::NamedTempFile,
+            Storage,
+            ContractAddress,
+            StarknetBlockHash,
+            StarknetBlockNumber,
+        ) {
+            let mut source_path = PathBuf::from(env!("CARGO_MANIFEST_DIR"));
+            source_path.push("fixtures/mainnet.sqlite");
+
+            let db_file = tempfile::NamedTempFile::new().unwrap();
```

Comment:
This will probably leak the `-wal` and `-shm` files created by SQLite.

You should either create a temporary directory instead of the file (so that all auxiliary files are removed as well) or not use WAL.